ddtrace/opentracer/tracer.py

```diff
@@ -49,8 +50,6 @@ def __init__(self, service_name=None, config=None, scope_manager=None):
                 str_invalid_keys = ','.join(invalid_keys)
                 raise ConfigException('invalid key(s) given (%s)'.format(str_invalid_keys))
 
-        # TODO: we should set a default reasonable `service_name` (__name__) or
-        # similar.
         if not self._service_name:
             raise ConfigException('a service_name is required')
```

Comment:
It's possible that service name is `None` even after the application name discovery, so I'd say to change the text in something similar to: `Cannot detect the 'service_name'. Please set the 'service_name=' keyword argument.`